test/cppsuite/src/util/barrier.h

```diff
@@ -0,0 +1,56 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef BARRIER_H
+#define BARRIER_H
+#include <mutex>
+#include <condition_variable>
+
+namespace test_harness {
+
+/*
+ * A barrier is a sychronization class that allows i_count threads to perform an action at the same
+ * time. They are introduced in C++20 so we need a basic implementation in the meantime.
+ *
+ * In order to synchronize across threads call wait() on each thread, once i_count threads have
+ * called wait() they will exit the wait() and continue.
+ */
+class barrier {
+    public:
+    explicit barrier(std::size_t i_count);
```

Comment:
I'd recommend adding 

`barrier(barrier const&) = default;`
`~barrier() = default;`

to clarify that the default copy constructor and destructor are correct.

Reply:
87c1d8abf